source/views/dictionary/list.js

```diff
@@ -77,21 +101,58 @@ export class DictionaryView extends React.Component {
     return <ListSeparator key={`${sectionId}-${rowId}`} />
   }
 
+  handleResults = (results: {[key: string]: Array<WordType>}) => {
+    this.setState({results: results})
+  }
+
+  checkIfSearching = (text: string) => {
+    return text === ''
+      ? this.setState({hideClear: true})
+      : this.setState({hideClear: false})
+  }
+
+  searchBar: SearchBar
+
   render() {
+    if (!terms) {
+      return <LoadingView />
+    }
+
     return (
-      <StyledAlphabetListView
-        data={groupBy(terms, item => head(item.word))}
-        cell={this.renderRow}
-        // just setting cellHeight sends the wrong values on iOS.
-        cellHeight={
-          rowHeight +
-            (Platform.OS === 'ios' ? 11 / 12 * StyleSheet.hairlineWidth : 0)
-        }
-        sectionHeader={this.renderHeader}
-        sectionHeaderHeight={headerHeight}
-        showsVerticalScrollIndicator={false}
-        renderSeparator={this.renderSeparator}
-      />
+      <View style={styles.wrapper}>
+        <SearchBar
+          ref={ref => this.searchBar = ref}
+          data={terms}
+          handleChangeText={text => this.checkIfSearching(text)}
```

Comment:
Please remove the arrow functions from these two: it's unnecessary overhead; because the arguments from the callbacks match up with the things you're calling, you can just pass references to the functions instead. Handleresults={this.handleResults}, for instance.